blockchain-agent-proof/src/main/java/com/sphereon/alfresco/blockchain/agent/proof/BlockchainProofConfigVerifier.java

```diff
@@ -1,78 +1,84 @@
-package com.sphereon.alfresco.blockchain.agent.backend.tasks.ondemand;
+package com.sphereon.alfresco.blockchain.agent.proof;
 
 import com.sphereon.sdk.blockchain.proof.api.ConfigurationApi;
 import com.sphereon.sdk.blockchain.proof.handler.ApiException;
 import com.sphereon.sdk.blockchain.proof.model.ChainSettings;
 import com.sphereon.sdk.blockchain.proof.model.CreateConfigurationRequest;
-import com.sphereon.alfresco.blockchain.agent.backend.tasks.Task;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Component;
 import org.springframework.util.Assert;
 
+import javax.annotation.Nonnull;
 import java.util.List;
 
 import static com.sphereon.sdk.blockchain.proof.model.ChainSettings.ContentRegistrationChainTypesEnum;
 import static com.sphereon.sdk.blockchain.proof.model.ChainSettings.HashAlgorithmEnum;
 import static com.sphereon.sdk.blockchain.proof.model.CreateConfigurationRequest.AccessModeEnum;
 
+/**
+ * Task that verifies that the Blockchain Proof config is valid. Annotated with {@code @ApplicationListener} to run as startup-time check.
```

Comment:
I must be missing the annotation ;)

Reply:
Haha, interesting. I think @sanderPostma  first experimented with an annotation and later implemented the `ApplicationListener` interface.

Reply:
My bad, I refactored this and merged some classes. So it should be an interface. Changed the comment.